compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -782,7 +782,34 @@ object RefChecks {
         report.error(problem(), clazz.srcPos)
       }
 
+      // check that basetype and subtype agree on types of trait parameters
+      //
+      // I.e. trait and class parameters not only need to conform to the expected
+      // type of the corresponding base-trait, but also to the type as seen by the
+      // inheriting subtype.
+      def checkTraitParametersOK() = for {
+        parent <- clazz.info.parents
+        parentSym = parent.classSymbol
+        if parentSym.isClass
+        cls = parentSym.asClass
+        if cls.paramAccessors.nonEmpty
+        param <- cls.paramAccessors
```

Comment:
paramAccessors contains both type and term parameters, is this actually needed for soundness or is checking types enough?

Reply:
What I try to do here is check the different types of the constructor argument from different perspectives. I am using `paramAccessors` since I saw it used in `checkParameterizedTraitsOK` above.
When you say "checking types enough", which types are you referring to?

Reply:
I mean, in the Scala 2 error for https://github.com/lampepfl/dotty/issues/11018#issuecomment-756777091, we get:
>  class F inherits different type instances of class C:
C[Foo] and C[Bar]

So it's checking the types, not the terms.

Reply:
Ok, I'll try to port the Scala 2 implementation as an alternative.